doc/source/cluster/ray-client.rst

```diff
@@ -27,27 +29,39 @@ By changing ``ray.init()`` to ``ray.init("ray://<host>:<port>")``, you can conne
    do_work.remote(2)
    #....
 
+Client arguments
+----------------
+
+Ray client is used when the address passed into ``ray.init`` is prefixed with the name of a protocol and ``://``. **This will only work with Ray 1.5+.** If you're using an older version of ray, see the `1.4.1 docs <https://docs.ray.io/en/releases-1.4.1/cluster/ray-client.html>`_. Currently, two protocols are built into ray by default:
 
-You can also connect using the ClientBuilder API, but this will eventually be deprecated.
+**Local mode** is used when the address is prefixed with ``local://``. If no address is provided, i.e. ``ray.init("local://")``), a local cluster is created and connected to directly by attaching a driver. If an address is provided, i.e. ``ray.init("local://localhost:6379")``, Ray will attempt to attach a driver to an existing cluster at the specified address. Local mode accepts all of the same keywords as ``ray.init`` and uses them to configure the local cluster.
```

Comment:
what is the difference between `ray.init("local://localhost: 6379")` over `ray.init("localhost: 6379")`
Also, in general "Local mode" is not clear (mainly it is not clear what local means).

Reply:
Also, are you sure it connects directly and not over the ray client code path?

Reply:
After syncing with Ian, it looks like this probably isn't even worth mentioning in the client docs (it doesn't even use the ray client, and there's not really a reason to use it over just ray.init()). Looks like it was more of a convenience thing for ray.client() so that you could just change the argument instead to ray.client().connect() instead of changing the entire thing to ray.init().

Reply:
Should we then remove the special handling of ‘local://‘?
Also, cc @pcmoritz to double verify.

Reply:
It'll be useful to keep the code for now, I think(?) the long term plan is to require any address passed into ray.init to start with something://, and ray.init(local://address) will just be used if users want to keep the current behavior of ray.init(address)

Reply:
Thanks @ckw017 for following up on this. I think it is better to remove code and then later add it if we need it (we have the git diff/commits in case we want to retrieve it).

Reply:
+1 to not support this for now, now that we have unified the ray.init API and the ClientBuilder API I don't think this functionality is really needed.

The idea here was to only support `ray.init("local://")` (without an address), which starts a local Ray cluster and connects to it via the Ray client. I don't think we should support `ray.init("local://address:port")` because that's already covered by `ray.init("ray://address")`. If we later want to add the functionality to start a local cluster and connect with the client, we may even want to do it with `ray.init("ray://")`.